src/metatensor/models/utils/data/dataset.py

```diff
@@ -28,15 +28,15 @@ def compiled_join(a, axis, remove_tensor_name):
     compiled_join = torch.jit.script(metatensor.torch.join)
 
 
-def get_all_species(dataset: _BaseDataset) -> List[int]:
+def get_all_species(datasets: List[_BaseDataset]) -> List[int]:
     """
-    Returns the list of all species present in the dataset.
+    Returns the list of all species present in a list of datasets.
 
     Args:
-        dataset: The dataset.
+        datasets: The datasets.
 
     Returns:
-        The list of species present in the dataset.
+        The list of species present in the datasets.
```

Comment:
they are sorted according to `set`, right?

`:returns: list of sorted species present in the datasets.`